### PR TITLE
Refactor datetime input component

### PR DESCRIPTION
**Pull request by frankfu03**
This PR is to refactor the datetime-input component, as the first step of merging date-input component into datetime-input component to form a combined input component.
Main changes made in this PR include: 
- Any changes made in datetime picker will automatically been updated on the datetime input box without clicking the "OK" button;
- Highlight time input boxes with red color if the values of housr, minutess, second in datetime picker are invalid;
- Removed the "OK" and "Cancel" buttons on datetime picker.
You may test the changes via DEV [site-manager](https://dev.gnss-site-manager.geodesy.ga.gov.au).

### CODE REVIEW

**Comment by lbodor** [comment]
Thanks for the link, it looks in dev.

File: src/client/app/shared/form-input/datetime-input.component.ts [hunk 1]
```diff
@@ -14,52 +14,39 @@ import { MiscUtils } from '../index';
     templateUrl: 'datetime-input.component.html',
     styleUrls: ['datetime-input.component.css']
 })
-export class DatetimeInputComponent extends AbstractInput implements OnInit, DoCheck {
-    @Input() dateType: string = 'Installed-Removed';
-    public datetime: string = '';
+export class DatetimeInputComponent extends AbstractInput implements OnInit {
+
+    @Input()
+    dateType: string = 'Installed-Removed';
```
Comment (lbodor) [inline src/client/app/shared/form-input/datetime-input.component.ts:20]: Is this what discriminates between date time and date only modes? I think booleans like withTime or onlyDate would be more precise.

**Comment by lbodor** [comment]
Will second step be to remove the date only component?

**Review by lbodor — APPROVED** [review]

File: src/client/app/shared/form-input/datetime-input.component.ts (hunk 1 above)
Comment (frankfu03) [reply src/client/app/shared/form-input/datetime-input.component.ts:20]: No, this boolean flag is to discriminate Date Installed/Removed (antenna, receiver, etc) and Start/End Date (Multipath Source, Signal Obstruction, etc).
For date time and date only modes, I used showTime boolean flag. onlyDate is also a good name for this flag.

Comment (lbodor) [reply src/client/app/shared/form-input/datetime-input.component.ts:20]: Thanks, I think `showTime` is perfect.

Comment (frankfu03) [reply src/client/app/shared/form-input/datetime-input.component.ts:20]: Thanks.

**Pull request merged**